src/ensembl/src/content/app/species-selector/components/species-autosuggestion-panel/SpeciesAutosuggestionPanel.tsx

```diff
@@ -4,15 +4,24 @@ import SpeciesSearchMatch from '../species-search-match/SpeciesSearchMatch';
 
 import styles from './SpeciesAutosuggestionPanel.scss';
 
-import { SearchMatches } from 'src/content/app/species-selector/types/species-search';
+import { SearchMatch } from 'src/content/app/species-selector/types/species-search';
 
 type Props = {
-  matches: SearchMatches;
+  matches: SearchMatch[];
+  onMatchSelected: (match: SearchMatch) => void;
 };
 
 const SpeciesAutosuggestionPanel = (props: Props) => {
+  const onMatchSelected = (match: SearchMatch) => () => {
+    props.onMatchSelected(match);
+  };
+
   const matches = props.matches.map((match) => (
-    <SpeciesSearchMatch match={match} key={match.description} />
+    <SpeciesSearchMatch
+      match={match}
+      onClick={onMatchSelected(match)}
```

Comment:
A bit curious about this one. Does using a wrapper function prevent re-renders that are usually caused by using inline functions?

Reply:
No, it doesn't :-) It was written solely to trick the linter, which at the moment is complaining when you write functions inline. This is a perfect example of why this linting rule is stupid, because performance-wise both ways of writing the handler are exactly the same (and the performance cost of generating a new function each time the component re-renders is minuscle anyway).

Here's my favourite Ryan Florence's tweet about unhealthy obsession with functions created during render:

![image](https://user-images.githubusercontent.com/6834224/54220630-be29f400-44e9-11e9-9bc4-0a4efc7613fa.png)

I suggest we get rid of that linting rule :-)

Reply:
There is an interesting discussion about it here: https://twitter.com/dan_abramov/status/1005566478089965574. Some are leaning towards not using them at all. Having said that, functional components will re-render the function anyway 😄

Reply:
> In practice I’d suggest ignoring this rule until you have real perf problems, and then maybe following it where it matters

Wise words! :smile: